### PR TITLE
Enforce gas price to be greater than base fee in block creation logic

### PR DESCRIPTION
**Pull request by abdelhamidbakhta**
## PR description
Add a check to verify that the gas price is greater than base fee for EIP-1559 blocks. Drop the transaction from the pool if not.

### CODE REVIEW

**Comment by RatanRSur** [comment]
Design question, why delete it in the transaction selector? My first instinct is that the transaction selector should only be responsible for selecting from the pool and the pool is responsible for evicting.

**Comment by abdelhamidbakhta** [comment]
> Design question, why delete it in the transaction selector? My first instinct is that the transaction selector should only be responsible for selecting from the pool and the pool is responsible for evicting.

Yeah I changed my mind because actually the user can override the behaviour of the pool but since we trust block we produce i ended up having invalid transactions in a mined block. I think I will add the check as well in the transaction pool to have more checks.

**Comment by abdelhamidbakhta** [comment]
> Design question, why delete it in the transaction selector? My first instinct is that the transaction selector should only be responsible for selecting from the pool and the pool is responsible for evicting.

And also there is some work about defining guidelines for transaction pool sorting / eviction under 1559
We will have to rework it anyway

File: ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java [hunk 1]
```diff
@@ -210,6 +213,16 @@ private TransactionSelectionResult evaluateTransaction(
...
     }
 
+    if (processableBlockHeader.getBaseFee().isPresent()
+        && Wei.of(processableBlockHeader.getBaseFee().get())
+                .compareTo(actualMinTransactionGasPriceInBlock)
+            > 0) {
+      LOG.error(
+          "transaction price must be greater than base fee, dropping transaction {} from the pool",
+          transaction.getHash().toHexString());
+      return TransactionSelectionResult.DELETE_TRANSACTION_AND_CONTINUE;
```
Comment (RatanRSur) [inline ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java:223]: Can we just `CONTINUE` here then? We already remove the lowest priority transaction in `addTransaction` in `PendingTransactions` so when the 1559 tx pool prioritization lands, that will automatically be taken care of.

**Pull request closed without merging**